std/range/package.d

```diff
@@ -9223,3 +9223,5 @@ if (Rs.length > 1 &&
     m.popFront;
     assert(m.empty);
 }
+
+package T* addressOf(T)(ref T val) { return &val; }
```

Comment:
Similar functions already exist in other parts of Phobos, namely `std/parallelism.d` and `std/algorithm/internal.d`. We should factor all of these out into a common place, instead of copy-n-pasting them into every module/package that needs to take an address of a property.